### PR TITLE
Home Task 09 Completed

### PR DESCRIPTION
**Pull request by jhonfmg7**
Home task 09 completed,
Project created with "create-react-app"
adding testing and merging with routing. Using jest, react-testing-library and cypress
Captura de pantalla 2022-12-22 a la(s) 11 57 08 a m
Captura de pantalla 2022-12-22 a la(s) 11 57 25 a m

### CODE REVIEW

File: cypress/e2e/crud.cy.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
+describe('proving CRUD functionalities for movies app', () => {
+  it('proving the URL have the correct queries', () => {
+    cy.visit('http://localhost:3000/');
+
+    cy.url()
```
Comment (mmarkelov) [inline cypress/e2e/crud.cy.ts:5]: Cannot it be just one expression?

File: src/__test__/Header.test.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    expect( headerSearchInput.tagName ).toBe("INPUT");
+    expect( headerSearchInput.value ).toBe('');
+
+    const headerSearchButton = getByTestId('header_search_button');
+    expect( headerSearchButton ).toBeInTheDocument();
+    expect( headerSearchButton.tagName ).toBe('BUTTON');
+
+    let searchParam = window.location.pathname.split('/')[2]
+    expect( searchParam ).toBe(undefined);
+    expect( searchParam ).not.toBe('whatever%20you%20would%20like%20to%20search');
```
Comment (mmarkelov) [inline src/__test__/Header.test.js]: maybe better?
```suggestion
    expect( searchParam ).not.toBe(encodeURIComponent('whatever you would like to search'));
```

File: src/components/main/navbar/index.tsx [hunk 3]
```diff
@@ -40,13 +40,13 @@ function NavBar() {
...
         { CATEGORIES.map((category, index) => (
-          <li key={index} className={searchParams.get("genre") === category ? styles.navbarItemActive : styles.navbarItem} onClick={() => setSearchParams({ genre: category, sortBy: sortBySelected })}>{ camelCase(category) }</li>
+          <li key={index} data-testid={`genre_option_${category}`} className={searchParams.get("genre") === category ? styles.navbarItemActive : styles.navbarItem} onClick={() => setSearchParams({ genre: category, sortBy: sortBySelected })}>{ camelCase(category) }</li>
         )) }
       </ul>
       <div className={styles.sortBySelect}>
         <p className={styles.secondaryText}>Sort by</p>
         <div className={styles.selectContainer}>
-          <select className={styles.selectInput} value={searchParams.get("sortBy")} onChange={(e) => setSearchParams({ genre: typeSelected, sortBy: e.target.value })}>
+          <select className={styles.selectInput} value={searchParams.get("sortBy") ? searchParams.get("sortBy") : ""} data-testid="sortby_select" onChange={(e) => setSearchParams({ genre: typeSelected, sortBy: e.target.value })}>
```
Comment (mmarkelov) [inline src/components/main/navbar/index.tsx]:
```suggestion
          <select className={styles.selectInput} value={searchParams.get("sortBy") ?? ""} data-testid="sortby_select" onChange={(e) => setSearchParams({ genre: typeSelected, sortBy: e.target.value })}>
```

File: src/components/modals/AddOrEditMovieForm/Input.tsx [hunk 4]
```diff
@@ -15,15 +15,19 @@ function Input(props: Input) {
...
       <input
         id={id}
+        data-testid={`new_movie_input_${value}`}
+        data-cy={`new_movie_input_${value}`}
         name={value}
         type={type}
         placeholder={placeholder}
         className={errors[value] && touched[value] ? styles.inputWithError : styles.input}
-        value={info[value]}
+        value={info[value] ? info[value] : ""}
```
Comment (mmarkelov) [inline src/components/modals/AddOrEditMovieForm/Input.tsx]:
```suggestion
        value={info[value] ?? ""}
```

File: src/components/modals/MoreInfoMovie.tsx [hunk 5]
```diff
@@ -16,15 +16,15 @@ import useCustomSearchParams from "../../hooks/useCustomSearchParams";
 
 interface Props {
     movie?: Movie,
-    setIsOpen: React.Dispatch<React.SetStateAction<boolean>>
+    setIsOpen: any
```
Comment (mmarkelov) [inline src/components/modals/MoreInfoMovie.tsx]: plz avoid `any`

**Comment by mmarkelov** [comment]
From my point of view - it's better to have more specific tests instead of general one

File: src/components/modals/MoreInfoMovie.tsx (hunk 5 above)
Comment (jhonfmg7) [reply src/components/modals/MoreInfoMovie.tsx]: Resolved

File: cypress/e2e/crud.cy.ts (hunk 1 above)
Comment (jhonfmg7) [reply cypress/e2e/crud.cy.ts:5]: I got it, thank you for the tip

File: src/__test__/Header.test.js (hunk 2 above)
Comment (jhonfmg7) [reply src/__test__/Header.test.js]: resolved

File: src/components/main/navbar/index.tsx (hunk 3 above)
Comment (jhonfmg7) [reply src/components/main/navbar/index.tsx]: resolved

File: src/components/modals/AddOrEditMovieForm/Input.tsx (hunk 4 above)
Comment (jhonfmg7) [reply src/components/modals/AddOrEditMovieForm/Input.tsx]: resolved

**Comment by jhonfmg7** [comment]
> From my point of view - it's better to have more specific tests instead of general one

**Pull request merged**